test/ein_reduce.cpp

```diff
@@ -177,5 +221,22 @@ TEST(einsum_sum_2d) {
   }
 }
 
+TEST(ein_reduce_max_2d) {
+  array_of_rank<int, 3> T({4, 5, 8});
+  fill_pattern(T);
+
+  // Reduce T along the i and k dimensions, keeping j.
+  auto max_ik = make_array<int>(make_shape(T.j()), std::numeric_limits<int>::min());
+  auto r = ein<j>(max_ik);
+  ein_reduce(r = max(r, ein<i, j, k>(T)));
```

Comment:
@jiawen :)

Maybe a `max_assign` operator would be nice, but this isn't bad either? And it's nice to be a bit more explicit.